### PR TITLE
Feature/support large dml split

### PR DESCRIPTION
**Pull request by newborn22**
## Related Issue(s)

#403 

## Checklist

-   [x] Tests were added or are not required
-   [x] Documentation was added or is not required

### CODE REVIEW

File: go/vt/vttablet/jobcontroller/controller.go [hunk 1]
```diff
@@ -0,0 +1,1890 @@
...
+
+						_, _ = jc.execQuery(ctx, tableSchema, fmt.Sprintf(sqlTemplateDropTable, jobBatchTable))
+					}
+				}
+			}
+		}
+
+		jc.tableMutex.Unlock()
+		time.Sleep(healthCheckInterval * time.Millisecond)
+	}
```
Comment (earayu) [inline go/vt/vttablet/jobcontroller/controller.go:1383]: Should use tick

File: go/vt/vttablet/jobcontroller/controller.go [hunk 2]
```diff
@@ -0,0 +1,1890 @@
...
+	pkPart := ""
+	for _, pkInfo := range pkInfos {
+		if !isFirstPk {
+			pkPart += ","
+		}
+		pkPart += pkInfo.pkName
+		isFirstPk = false
+	}
+	//1.2.生成select pk的sql
+	batchSplitSelectSQL := strings.Replace(batchCountSQL, "count(*) as count_rows", pkPart, 1) + "order by " + pkPart
```
Comment (earayu) [inline go/vt/vttablet/jobcontroller/controller.go:815]: Should not use strings to manipulate SQL, use sqlparser

File: go/vt/vttablet/jobcontroller/controller.go [hunk 3]
```diff
@@ -0,0 +1,1890 @@
...
+	// 定义正则表达式，匹配"( (greatThanPart) AND (lessThanPart) )"
+	curBatchGreatThanPart := ""
+	newBatchLessThanPart := ""
+
+	// 这个正则表达式用于匹配出"( (greatThanPart) AND (lessThanPart) )"greatThanPart和lessThanPart，也就是每条batch sql中用于限定PK范围的部分
+	regexPattern := `\(\s*\((.*)\)\s*AND\s*\((.*)\)\s*\)`
+
+	// 编译正则表达式
+	regex := regexp.MustCompile(regexPattern)
+
```
Comment (earayu) [inline go/vt/vttablet/jobcontroller/controller.go:878]: use sqlparser🤔

**Comment by earayu** [comment]
Need more Unit Tests.
For adding more Unit Tests, you may need to refactor some functions.

File: go/vt/sqlparser/ast_format.go [hunk 4]
```diff
@@ -311,6 +311,53 @@ func (node *AlterMigration) Format(buf *TrackedBuffer) {
...
+	case CancelDMLJobType:
+		alterType = "cancel"
+	case CancelAllDMLJobType:
+		alterType = "cancel all"
+	case PauseDMLJobType:
+		alterType = "pause"
+	case PauseAllDMLJobType:
+		alterType = "pause all"
+	case ResumeDMLJobType:
+		alterType = "resume"
```
Comment (gerayking) [inline go/vt/sqlparser/ast_format.go:335]: should be taken to reduce the use of magic strings

File: go/vt/vttablet/jobcontroller/controller.go [hunk 5]
```diff
@@ -0,0 +1,987 @@
...
+				jc.initDMLJobRunningMeta(runnerArgs.uuid, runnerArgs.table)
+			}
+		}
+
+		jc.workingTablesMutex.Unlock()
+		jc.tableMutex.Unlock()
+	}
+}
+
+func (jc *JobController) tableGC(ctx context.Context, uuid, tableSchema, batchInfoTable, statusSetTime string) error {
```
Comment (earayu) [inline go/vt/vttablet/jobcontroller/controller.go:829]: You should refer tableGC module.

**Review by earayu — APPROVED** [review]
LGTM

**Pull request merged**